src/atoms/Button/Button.web.js

```diff
@@ -1,20 +1,454 @@
+        return 'flex-start';
+    }
+  },
+  letterSpacing({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'medium';
+      default:
+        return 'small';
+    }
+  },
+  spaceBetween({ size, iconAlign }) {
+    switch (size) {
+      case 'xsmall':
+        if (iconAlign === 'left') {
+          return [0, 0.5, 0, 0];
+        }
+        return [0, 0, 0, 0.5];
+      case 'small':
+        if (iconAlign === 'left') {
+          return [0, 0.5, 0, 0];
+        }
+        return [0, 0, 0, 0.5];
+      case 'medium':
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+      case 'large':
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+      default:
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+    }
+  },
+  lineHeight({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'xsmall';
+      case 'small':
+        return 'small';
+      case 'medium':
+        return 'medium';
+      case 'large':
+        return 'medium';
+      default:
+        return 'medium';
+    }
+  },
+};
+
+const StyledButton = styled.button`
+  &&& {
+    background-color: ${styles.backgroundColor};
+    border-radius: ${(props) => props.theme.spacings.xxsmall};
+    border: ${styles.border};
+    width: ${(props) => (props.block ? '100%' : '')};
+    outline: none;
+    &:hover {
+      border: ${styles.hoverBorder};
+      background-color: ${styles.hoverBackgroundColor};
+    }
+    &:focus {
+      border: ${styles.focusBorder};
+      background-color: ${styles.focusBackgroundColor};
+    }
+    &:active {
+      border: ${styles.activeBorder};
+      background-color: ${styles.activeBackgroundColor};
+    }
+  }
+`;
+
+const Button = ({
+  onClick,
+  children,
+  variant,
+  size,
+  disabled,
+  variantColor,
+  icon,
+  iconAlign,
+  align,
+  block,
+}) => {
+  const theme = useContext(ThemeContext);
+
+  const color = `${variantColor || 'primary'}`;
+
+  if (typeof children !== 'string' && typeof children !== 'undefined') {
+    throw new Error(
+      `Error in Button: expected \`children\` of type \`string\` but found ${typeof children}`,
+    );
+  }
+
+  return (
+    <Flex
+      flex={block ? 1 : 0}
+      flexDirection="row"
+      alignItems="center"
+      alignSelf={styles.align({ align })}
+      justifyContent="center"
+    >
+      <Size height={styles.height({ size, theme })}>
+        <Space padding={styles.padding({ size, children, theme })}>
+          <StyledButton
+            color={color}
+            onClick={onClick}
+            variantColor={variantColor}
+            disabled={disabled}
+            size={size}
+            block={block}
+            variant={variant}
+            {...automation('ds-button')}
+          >
+            <React.Fragment>
+              {icon && iconAlign === 'left' && (
```

Comment:
```suggestion
              {icon && iconAlign === 'left' ? (
```

Reply:
Please don't use `&&` for conditional rendering. Use `&&` for checks and `?` for conditional render. Make change at other places as well.

Reply:
added

Reply:
Noice, thanks!